data_diff/database.py

```diff
@@ -186,6 +287,12 @@ def md5_to_int(self, s: str) -> str:
     def to_string(self, s: str):
         return f"{s}::varchar"
 
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
```

Comment:
The relationship between the 'compiler' in `sql.py` and what's in the Database drivers themselves is starting to blur a bit for me 👀 

Do you think it'd be worth moving some of this driver-specific logic increasingly in there, so you can call e.g. `Normalize()` with a Database as another argument?

Reply:
> starting to blur a bit for me 

Yeah, I agree. It's just a lot of db-specific code, and very few generic SQL code that works everywhere. It makes it feel like the design is wrong. But it's the databases that are wrong ;)

> Do you think it'd be worth moving some of this driver-specific logic increasingly in there,

I'll do what I can, but I think it's inevitable that Database will provide quite a bit of db-specific SQL code

> Normalize() with a Database as another argument

I don't think it's a good idea. You'd never use `[ Normalize(x, db1), Normalize(x, db2) ]`, so we shouldn't make it possible to express.

Reply:
Ok, we can discuss later if we need to change the design a bit. I think this'll happen more and more with the other types.. 😅